packages/core/strapi/lib/types/utils/object.d.ts

```diff
@@ -10,10 +10,13 @@
  * type X = KeysBy<Obj, Base>
  * // 'foo' | 'bar'
  */
-export type KeysBy<TValue, TTest> = string &
-  {
-    [key in keyof TValue]: TValue[key] extends TTest ? key : never;
-  }[keyof TValue];
+export type KeysBy<TValue, TTest> = {
+  [key in keyof TValue]: TValue[key] extends TTest ? key : never;
+}[keyof TValue];
+
+export type KeysNotBy<TValue, TTest> = {
```

Comment:
Could you add a comment with a description here (just like for the KeysBy)?

Reply:
What do you think about `KeysExcept`? Not sure about either one of them.

Reply:
How about `OmitWhere` ?

Reply:
That sounds nice, but it would be to retrieve properties. Here we just want the keys

Reply:
lgtm

Reply:
I used `KeysExcept` because there is a non documented convention with "By" in all the code base that we are already following, so I'm not sure if changing this now would confuse us later on.